index.ts

```diff
@@ -10,150 +42,191 @@ async function getData(url: string) {
+                    ','
+                )}&fields=name&fields=borders&fields=cca3`
+            );
+            if (!res.ok) {
+                throw new Error(res.statusText);
             }
-            if (unvisited!.length > 0) {
-                /* eslint-disable no-await-in-loop */
-                await getPath(unvisited);
+            const data = await res.json();
+            fetchCount += 1;
+            if (data[0].borders.length === 0) {
+                return previous;
             }
+        } catch (err) {
+            console.log(err);
         }
-    }
 
-    function findShortestPath(fromCountry: string, toCountry: string, map: TGraph) {
-        function bfs2(fromCountry: string, map: TGraph) {
-            const queue = [fromCountry];
-            const visited = { [fromCountry]: 1 };
-            const previous = { [fromCountry]: '' };
+        async function handleVertex(countryCodes: Array<string>): Promise<void> {
+            const isSomeCountryUnvisited = countryCodes.some((v) => !Object.keys(map).includes(v));
+            if (!map[toCountry] && isSomeCountryUnvisited) {
+                let countryData;
+                try {
+                    const res = await fetch(
+                        `https://restcountries.com/v3.1/alpha?codes=${countryCodes.join(
+                            ','
+                        )}&fields=name&fields=borders&fields=cca3`
+                    );
+                    if (!res.ok) {
+                        throw new Error(res.statusText);
+                    }
+                    countryData = await res.json();
+                } catch (err) {
+                    console.log(err);
+                }
+
+                fetchCount += 1;
+
+                for (let i = 0; i < countryData.length; i++) {
+                    if (!allBorders[countryData[i].cca3]) {
+                        allBorders[countryData[i].cca3] = {
+                            borders: countryData[i].borders,
+                            name: countryData[i].name.common,
+                        };
+                        // если нашли страну, выходим из цикла
+                        if (countryData[i].cca3 === toCountry) {
+                            queue = [];
+                            return;
+                        }
 
-            function handleVertex(vertex: string) {
-                const neighboursList = map[vertex].borders;
+                        const neighboursList = countryData[i].borders;
+                        queue.push(neighboursList);
 
-                neighboursList.forEach((neighbour) => {
-                    if (!visited[neighbour]) {
-                        visited[neighbour] = 1;
-                        queue.push(neighbour);
-                        previous[neighbour] = vertex;
+                        for (const neighbour of neighboursList) {
+                            if (!visited[neighbour]) {
+                                visited[neighbour] = 1;
```

Comment:
```suggestion
                                visited[neighbour] = true;
```
Чего буквы экономить, с 1 выглядит как какой-то счётчик, ищешь где становится 2.